src/views/workflow-history/workflow-history-event-status-badge/workflow-history-event-status-badge.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+import { Spinner } from 'baseui/spinner';
+import { MdCheck, MdHourglassTop, MdReportGmailerrorred } from 'react-icons/md';
+
+import {
+  WORKFLOW_EVENT_STATUS,
+  WORKFLOW_EVENT_STATUS_BADGE_SIZES,
+  iconSizeMap,
+} from './workflow-history-event-status-badge.constants';
+import { styled } from './workflow-history-event-status-badge.styles';
+import type { Props } from './workflow-history-event-status-badge.types';
+
+export default function WorkflowHistoryEventStatusBadge({
+  status,
+  size = WORKFLOW_EVENT_STATUS_BADGE_SIZES.medium,
+}: Props) {
+  if (!WORKFLOW_EVENT_STATUS[status]) return null;
+
+  const renderIcon = () => {
+    const iconSize = iconSizeMap[size];
+    if (iconSize === undefined) return null;
```

Comment:
Nit: instead of using a map for icon sizes, you can define two constants (one for small and one for medium) and use a ternary so that you can avoid having the (unnecessary) null check.

Reply:
The map is to make sure adding new size to enum eg. large without adding iconSize would show an error